### PR TITLE
Backport PR #43455 on branch 1.3.x (REGR: fillna on f32 column raising for f64)

### PR DESCRIPTION
**Pull request by meeseeksmachine**
Backport PR #43455: REGR: fillna on f32 column raising for f64

### CODE REVIEW

**Comment by phofl** [comment]
cc @mzeitlin11

**Comment by jreback** [comment]
@mzeitlin11 i think need to update the fixture here: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=65861&view=logs&j=f39ddbca-85f6-5283-3bfe-05832bcf5208&t=60a5c9af-5075-529a-e870-f4d886d8e7ce

as recently changed the names :-< (note I think you *can* just push commits to this PR)

**Comment by mzeitlin11** [comment]
> @mzeitlin11 i think need to update the fixture here: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=65861&view=logs&j=f39ddbca-85f6-5283-3bfe-05832bcf5208&t=60a5c9af-5075-529a-e870-f4d886d8e7ce
> 
> as recently changed the names :-< (note I think you _can_ just push commits to this PR)

Thanks, will do

File: pandas/tests/series/methods/test_fillna.py [hunk 1]
```diff
@@ -702,6 +702,23 @@ def test_fillna_categorical_raises(self):
         with pytest.raises(TypeError, match=msg):
             ser.fillna(DataFrame({1: ["a"], 3: ["b"]}))
 
+    @pytest.mark.parametrize("dtype", [float, "float32", "float64"])
+    @pytest.mark.parametrize("fill_type", tm.ALL_REAL_NUMPY_DTYPES)
```
Comment (mzeitlin11) [inline pandas/tests/series/methods/test_fillna.py:706]:
```suggestion
    @pytest.mark.parametrize("fill_type", tm.ALL_NUMPY_DTYPES)
```

**Comment by phofl** [comment]
thx @mzeitlin11

**Pull request merged**